src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+                    string typeMetadata = string.Empty;
+                    string formatMetadata = string.Empty;
+                    if (dbObject.SourceDefinition.Columns.TryGetValue(backingColumnValue, out ColumnDefinition? columnDef) && columnDef is not null)
+                    {
+                        typeMetadata = SystemTypeToJsonDataType(columnDef.SystemType).ToString().ToLower();
+                    }
+
+                    properties.Add(field, new OpenApiSchema()
+                    {
+                        Type = typeMetadata,
+                        Format = formatMetadata
+                    });
+                }
+            }
+
+            OpenApiSchema schema = new()
+            {
+                Type = "object",
+                Properties = properties
+            };
+
+            return schema;
+        }
+
+        /// <summary>
+        /// Creates the default collection of responses for all requests in the OpenAPI
+        /// description document.
+        /// The OpenApiResponses dictionary key represents the integer value of the HttpStatusCode,
+        /// which is returned when using Enum.ToString("D").
+        /// The "D" format specified "displays the enumeration entry as an integer value in the shortest representation possible."
+        /// </summary>
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/standard/base-types/enumeration-format-strings#d-or-d"/>
+        /// <returns>Collection of default responses (400, 401, 403, 404).</returns>
+        private static OpenApiResponses CreateDefaultOpenApiResponses()
+        {
+            OpenApiResponses defaultResponses = new()
+            {
+                { HttpStatusCode.BadRequest.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.BadRequest)) },
+                { HttpStatusCode.Unauthorized.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Unauthorized)) },
+                { HttpStatusCode.Forbidden.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.Forbidden)) },
+                { HttpStatusCode.NotFound.ToString("D"), CreateOpenApiResponse(description: nameof(HttpStatusCode.NotFound)) }
+            };
+
+            return defaultResponses;
+        }
+
+        /// <summary>
+        /// Converts the CLR type to JsonDataType
+        /// to meet the data type requirement set by the OpenAPI specification.
+        /// The value returned is formatted for the OpenAPI spec "type" property.
+        /// </summary>
+        /// <param name="type">CLR type</param>
+        /// <seealso cref="https://spec.openapis.org/oas/v3.0.1#data-types"/>
+        /// <returns>Formatted JSON type name in lower case: e.g. number, string, boolean, etc.</returns>
+        private static string SystemTypeToJsonDataType(Type type)
+        {
+            JsonDataType openApiTypeName = type.Name switch
+            {
+                "String" => JsonDataType.String,
```

Comment:
Instead of having constants here: can we use something like this: https://github.com/Azure/data-api-builder/blob/f6c11d8b08d8c95cef7d7cf2e45e30f8aed96d48/src/Service/Services/DbTypeHelper.cs#L15?

Reply:
Maybe we can rename the DbTypeHelper to a more generic class like TypeHelper and create another dictionary for this.

Reply:
done and added a dictionary in a generic typehelper class that achieves this.